### PR TITLE
test: Add additional integration tests

### PR DESCRIPTION
**Pull request by timfish**
- Adds 6 missing integration tests from the old test suite covering edge cases and bundler-specific functionality
- Adds CJS specific tests
- `readAllFiles` was modified to recursively read from subdirectories

### CODE REVIEW

File: packages/integration-tests-next/fixtures/vite4/basic-cjs.config.cjs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+const { sentryVitePlugin } = require("@sentry/vite-plugin");
+const { defineConfig } = require("vite");
+const { sentryConfig } = require("../configs/basic.config.js");
```
[the remark this answers is not in the document]
Comment (timfish) [reply packages/integration-tests-next/fixtures/vite4/basic-cjs.config.cjs:3]: The tests pass so there is no runtime error...

File: packages/integration-tests-next/fixtures/rolldown/basic-cjs.config.cjs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+const { sentryRollupPlugin } = require("@sentry/rollup-plugin");
+const { defineConfig } = require("rolldown");
+const { sentryConfig } = require("../configs/basic.config.js");
```
[the remark this answers is not in the document]
Comment (timfish) [reply packages/integration-tests-next/fixtures/rolldown/basic-cjs.config.cjs:3]: The tests pass so there is no runtime error...

**Review by chargome — APPROVED** [review]
Nice!

**Review by JPeer264 — APPROVED** [review]
LGTM. Haven't went into detail to every test though, but went over single ones in detail and they looked good
